CONTRIBUTING.md

```diff
@@ -13,70 +13,71 @@ Here are some ways to contribute to the project, from easiest to most difficult:
-7. **Merge and release!**
+### Review
 
-### Code review guidelines
+It’s important that every piece of code in Apollo packages is reviewed by at least one core contributor familiar with that codebase. If you want to expedite the code being merged, try to review your own code first! Here are some things we look for:
 
-It’s important that every piece of code in Apollo packages is reviewed by at least one core contributor familiar with that codebase. Here are some things we look for:
-
-1. **Required CI checks pass.** This is a prerequisite for the review, and it is the PR author's responsibility. As long as the tests don’t pass, the PR won't get reviewed.
-2. **Simplicity.** Is this the simplest way to achieve the intended goal? If there are too many files, redundant functions, or complex lines of code, suggest a simpler way to do the same thing. In particular, avoid implementing an overly general solution when a simple, small, and pragmatic fix will do.
+1. **All GitHub checks pass.** This is a prerequisite for the review, and it is the PR author's responsibility. The PR will not be reviewed until all the author has signed the Apollo CLA and all tests pass.
+2. **Simplicity.** Is this the simplest way to achieve the intended goal? If there are too many files, redundant functions, or complex lines of code, suggest a simpler way to do the same thing. In particular, avoid implementing an overly general solution when a simple, small, and pragmatic fix will do. Please also note that large pull requests take additional time to review. If your PR could be broken down into several smaller, more focused changes, please do that instead.
 3. **Testing.** Do the tests ensure this code won’t break when other stuff changes around it? When it does break, will the tests added help us identify which part of the library has the problem? Did we cover an appropriate set of edge cases? Look at the test coverage report if there is one. Are all significant code paths in the new code exercised at least once?
 4. **No unnecessary or unrelated changes.** PRs shouldn’t come with random formatting changes, especially in unrelated parts of the code. If there is some refactoring that needs to be done, it should be in a separate PR from a bug fix or feature, if possible.
 5. **Code has appropriate comments.** Code should be commented for *why* things are happening, not *what* is happening. *What* is happening should be clear from the names of your functions and variables. This is sometimes called "self-documenting code", but you may still need to add comments to explain why a workaround is necessary so other developers can better understand your code.
 6. **Idiomatic use of the language.** Make sure to use idiomatic Swift when working with this repository. We don't presently use SwiftLint or any other linter, but please use your common sense and follow the style of the surrounding code.
+
+## Testing
```

Comment:
@AnthonyMDev, this is the additional section about testing. It's intended to be brief. If we want to go into more detail about testing specifically then we should consider adding a TESTING.md file that can go into much more depth than we have space for here.

Reply:
I think this is all good information. What I was looking for was more of a technical explanation/guide of how to actually use this stuff. Maybe a new Testing.md file would be worth considering. It could include:
- Step-by-step on how to run the servers required for running integration tests locally.
- How to use the scripts we have to run code gen over both the in-project examples and the `TestCodeGenConfigurations`.
- How our test plans are organized.
- What the different jobs on Circle CI actually do
- etc.

It might be better to hold off on that until we do the repo/project restructuring, as it is subject to change.

Reply:
Yes, if we want that level of detail then a separate focused file would be the better place for it.

Reply:
I created https://github.com/apollographql/apollo-ios/issues/3040 for us to track it.